src/Validator/Validators/hdtXSDValidator.cpp

```
@@ -0,0 +1,316 @@
+			return;
+		}
+
+		if (tc.base == TypeConstraint::Base::Float || tc.base == TypeConstraint::Base::Integer) {
+			double dval = 0.0;
+			const auto [ptr, ec] = std::from_chars(val.data(), val.data() + val.size(), dval);
+			if (ec != std::errc{} || ptr != val.data() + val.size()) {
+				const std::string typeName = (tc.base == TypeConstraint::Base::Float) ? "float" : "integer";
+				ctx.addViolationWithCurrentPath(row, col, where + " has invalid " + typeName + " value '" + val + "'");
+				return;
+			}
+			if (tc.hasMin && dval < tc.minInclusive)
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)